packages/core/src/pages/[slug]/p.tsx

```diff
@@ -214,9 +214,13 @@ export const getStaticProps: GetStaticProps<
 
   const { data, errors = [] } = searchResult
 
+  const hasPrice = data?.product?.offers.offers.some(
```

Comment:
I wonder if we should move this logic to the `faststore/api`, throwing an `isNotFoundError` from there.